operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -669,6 +671,27 @@ private void internalMakeRightDomainPresence() {
             willInterrupt);
     }
 
+    private void populateServerStateHealthFromDomain(Packet packet) {
```

Comment:
Aren't we already updating these maps in ServerStatusReader? Why copy the logic to here?

Reply:
The domain status contains DomainConditions and ServerStatus. The ServerStatus contains the WebLogic servers' state and health information, which is updated based on what is in the two maps (serverState and serverHealth) in the packet. The domain conditions are only handled by the domain status update steps in DomainStatusUpdater. 

Prior to the code change in this PR, the server status update is already triggered from two code paths: make-right code path and ServerStatusReader/update  code path (see the places where the DomainStatusUpdater.createStatusUpdateStep is called). If I understand the logic correctly, the update in the make-right is to record possible cluster scale-up and scale-down changes. The update in the ServerStatusReader is to record state/health changes on the WebLogic server processes. The two code paths do NOT share the same packet instances and the contents. The ServerStatusReader populates the maps based on the responses that the reader gets, and add them to the packet. The make-right code does not populate the maps,  and therefore no maps are added to the packet. As a result, when the operator does a full recheck of an idle domain (running but no activities), the make-right code path will update the domain status to remove the server state and health, and the next update from the ServerStatusReader will update the states and health again, and put the actual state and health back. This behavior will repeat due to the periodical ServerStatusReader server status update logic and make-right full recheck logic, thus the domain version number keeps to be bumped up.

The solution in this PR is to populate the packet in the make-right code path with the current known state/health that the operator gets from the domain object, so that the logic starts with the correct initial server status. 

I agree that it is hard to unit test the combined behavior and end results from the code paths. I have been trying to find a way to test only the make-right code path instead, for example checking if the maps in the packet are populated correctly.

Reply:
Could the make-right code path simply leave the per-server status alone by copying the content from the current status, if any?

Reply:
A better solution, then, would probably be to encapsulate the logic, possibly in the ServerStatus itself. Both paths could then use that logic, and it could be unit-testable.

Reply:
@rjeberhard That is the change in the PR does; get the current status from the domain object. We need to populate the maps in the packet with those status contents because the remaining logic around this uses that information to determine the behavior.

Reply:
@russgold are you suggesting that both code paths read the status from the servers, and populate the maps with the responses?  Readers reaches out to each servers to get its state and health information, and I am not sure if we would want to do that in the make-right code path.

Reply:
I'm suggesting that both paths do what they need to do, but that the logic be encapsulated in ServerStatus, where it can be unit tested.